modules/apps/commerce/commerce-product-definitions-web/src/main/resources/META-INF/resources/edit_attachment_file_entry.jspf

```diff
@@ -97,29 +97,32 @@
 </commerce-ui:side-panel-content>
 
 <aui:script>
-	function getMetalJsFormData(metalJsForm) {
-		if (!metalJsForm) {
-			return [];
-		}
-
-		var renderer = Object.values(metalJsForm.refs)[0];
-
-		return Object.values(renderer.refs).map(function (option) {
-			return {
-				key: option.fieldName,
-				value: option.value,
-			};
-		});
-	}
+	var fieldValues = [];
 
 	Liferay.componentReady('ProductOptions<%= cpDefinitionId %>').then(function (
 		ddmForm
 	) {
 		if (!ddmForm.on) {
 			return;
 		}
-		ddmForm.on('fieldEdited', function () {
-			var fieldValues = getMetalJsFormData(ddmForm);
+		ddmForm.on('fieldEdited', function (e) {
+			var key = e.fieldInstance.fieldName;
+			var updatedItem = {
+				key: e.fieldInstance.fieldName,
+				value: e.value,
+			};
+			var itemFound = fieldValues.find(function (item) {
+				return item.key === key;
+			});
```

Comment:
We can add a new line here since its not alphabetized with the other variables.
```suggestion
			};

			var itemFound = fieldValues.find(function (item) {
				return item.key === key;
			});
```